test/mo-compiler-test.js

```diff
@@ -1,7 +1,7 @@
 import * as chai from 'chai';
 import { promisify } from 'util';
 import path from 'path';
-import { mo } from '../index.js';
+import { mo } from '../src/index.js';
```

Comment:
I think a good separate PR would be: move everything to `/src` + update package.json with the .npmignore config. It would remove half the changes from the current PR.

Reply:
ok about this, but i need to change a bit the way we are using ts... before was used as a bundler but in this case we only need it for the type check. 

I must agree with you, the modification of `src` should be done at a later stage, perhaps by replacing babel completely if you are fine with that